MotionPhotoReader/app/src/test/java/com/google/android/libraries/motionphotoreader/BufferProcessorTest.java

```diff
@@ -53,10 +73,48 @@ public void setUp() {
         availableOutputBuffers.offer(bundle);
     }
 
+    /**
+     * Creates a mock media format object. Used to represent video formats of motion photo files.
+     * @param width The width of the motion photo video, in pixels.
+     * @param height The height of the motion photo video, in pixels.
+     * @param duration The duration of the motion photo video, in microseconds.
+     * @param rotation The rotation applied to the motion photo, in degrees.
+     * @param mime The mime type of the motion photo video track.
+     * @return a mock MediaFormat object.
+     */
+    private MediaFormat createFakeVideoFormat(int width,
+                                              int height,
+                                              long duration,
+                                              int rotation,
+                                              String mime) {
+        MediaFormat videoFormat = mock(MediaFormat.class);
+        doAnswer((Answer<Integer>) invocation -> width)
+                .when(videoFormat)
+                .getInteger(eq(MediaFormat.KEY_WIDTH));
+        doAnswer((Answer<Integer>) invocation -> height)
+                .when(videoFormat)
+                .getInteger(eq(MediaFormat.KEY_HEIGHT));
+        doAnswer((Answer<Long>) invocation -> duration)
+                .when(videoFormat)
+                .getLong(eq(MediaFormat.KEY_DURATION));
+        doAnswer((Answer<Integer>) invocation -> rotation)
+                .when(videoFormat)
+                .getInteger(eq(MediaFormat.KEY_ROTATION));
+        doAnswer((Answer<String>) invocation -> mime)
+                .when(videoFormat)
+                .getString(eq(MediaFormat.KEY_MIME));
+
```

Comment:
doAnswer for method calls that need a side effect other than returning values. For these cases, it'll be easier to read as:

```
when(videoFormat.getString(MediaFormat.KEY_WIDTH)).thenReturn(width);
when(videoFormat.getString(MediaFormat.KEY_HEIGHT)).thenReturn(height);
...  
when(videoFormat.getString(MediaFormat.KEY_MIME)).thenReturn(mime);
```

Note eq(...) is implied and isn't needed unless one of the other args in the method call is a pattern matcher